middle_end/flambda2/to_cmm/to_cmm_static.ml

```diff
@@ -206,18 +231,29 @@ let immutable_unboxed_float32_array env res updates ~symbol ~elts =
 let static_const0 env res ~updates (bound_static : Bound_static.Pattern.t)
     (static_const : Static_const.t) =
   match bound_static, static_const with
-  | Block_like s, Block (tag, _mut, fields) ->
+  | Block_like s, Block (tag, mut, shape, fields) ->
+    (match mut with
+    | Immutable | Immutable_unique -> ()
+    | Mutable ->
+      (* CR mshinwell: actually we could probably permit updates on the flat
+         suffix even now *)
+      Misc.fatal_errorf
+        "Symbol %a: the GC does not currently support mutable fields in \
+         statically-allocated values"
+        Symbol.print s);
```

Comment:
Is that really true ? The GC already handles cases where dynamically allocated values are stored in the fields of a statically allocated value (these statically-allocated blocks are registered as roots so that the fields containing dynamically allocated values are correctly updated when the GC moves values).

So I'm wondering what prevents mutables fields in statically-allocated values ?

Reply:
I believe this is still the case, but I don't know the details.  I remember that Damien was going to write, or did write most of, a patch for runtime4 several years ago to lift this restriction but I'm not sure what came of it.  It would require rewriting for runtime5 I would guess.